spec/mongo/index/view_spec.rb

```diff
@@ -196,7 +196,70 @@
         expect(view.get('random_name')).to_not be_nil
       end
     end
-  end
+
+    context 'when providing an invalid partial index filter' do
+
+      let(:spec) do
+        { random:1 }
+      end
+
+      it 'raises an exception' do
+        expect {
+          view.create_one(spec, partial_filter_expression: 5)
+        }.to raise_error(Mongo::Error::OperationFailure)
+      end
+      it 'raises an exception' do
+        expect {
+          view.create_one(spec, partial_filter_expression: {"x": {"$asda": 3}})
+        }.to raise_error(Mongo::Error::OperationFailure)
+      end
+      it 'raises an exception' do
+        expect {
+          view.create_one(spec, partial_filter_expression: {
+                                  "$and": [{"$and": [{"x": {"$lt": 2}},
+                                                     {"x": {"$gt": 0}}]},
+                                           {"x": {"$exists": true}}]})
+        }.to raise_error(Mongo::Error::OperationFailure)
+      end
+    end
+
+    context 'when providing a valid partial index filter' do
+      let!(:result) do
+        view.create_one({"x": 1}, partial_filter_expression: {
+                                    "a": {"$lte": 1.5}})
+      end
+
+      it 'returns ok' do
+        expect(result).to be_successful
+      end
+
+      let!(:explain) do
+        authorized_collection.find(:x => 6, :a => 1).explain
+      end
+
+      it 'returns ok' do
```

Comment:
This should have a more descriptive test description instead of "returns ok".
Something like "Applies the filter when using the index""